contracts/contracts/GatewayDiamond.sol

```diff
@@ -73,6 +73,19 @@ contract GatewayDiamond {
         // The startConfiguration number is also 1 to match with nextConfigurationNumber, indicating we have
         // empty validator change logs
         s.validatorsTracker.changes.startConfigurationNumber = LibStaking.INITIAL_CONFIGURATION_NUMBER;
+        // Add genesis validators to the parent validators tracker.
+        uint256 vLength = params.genesisValidators.length;
+        for (uint256 i; i < vLength; ) {
+            address addr = params.genesisValidators[i].addr;
+            uint256 amount = params.genesisValidators[i].weight;
+            bytes memory metadata = params.genesisValidators[i].metadata;
+            LibValidatorSet.setMetadataFromConstructor(s.validatorsTracker.validators, addr, metadata);
+            LibValidatorSet.recordDeposit(s.validatorsTracker.validators, addr, amount);
```

Comment:
this technically works for `Staking` backed power, but will not work for federated. Ideally it would call record federated based on the permission mode.